src/electron/views/automated-checks/components/maximize-restore-button.tsx

```diff
@@ -0,0 +1,24 @@
+import { ActionButton } from 'office-ui-fabric-react/lib/Button';
+import * as React from 'react';
+
+import { RestoreIcon } from 'common/icons/restore-icon';
+import { NamedFC } from 'common/react/named-fc';
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+export interface MaximizeRestoreButtonProps {
+    isMaximized: boolean;
+    onClick: () => void;
+}
+
+export const MaximizeRestoreButton = NamedFC<MaximizeRestoreButtonProps>('TitleBar', (props: MaximizeRestoreButtonProps) => {
+    const content = props.isMaximized ? null : <RestoreIcon />;
```

Comment:
`ActionButton` also has an `onRenderIcon` prop you could assign `RestoreIcon` to; would that be better? I'm not whether it makes a difference but I'm suspicious about why it'd be provided if there was no difference

Reply:
It looks like the only difference in outcome is that `onRenderIcon` [passes in the button's props and renders before the button's text](https://github.com/OfficeDev/office-ui-fabric-react/blob/785a8733bedc868baa1a4ee17c0b551168cc710d/packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx#L283), so there would be no difference in this situation. If there were also text associated with the button, it would impact the location of the icon (before the text if `onRenderIcon` was used, after if it was passed as a child). Given that this is supposed to behave like an icon, I think `onRenderIcon` makes more sense, even if there's no difference in this context

Reply:
Updated